inference/synapse/synapse.go

```diff
@@ -119,23 +121,36 @@ func (s *Synapse) Close() {
 	log.Info("Synapse Engine Closed")
 }
 
-func (s *Synapse) InferByInfoHash(modelInfoHash, inputInfoHash string) ([]byte, error) {
+func CVMVersion(config *params.ChainConfig, num *big.Int) int {
+	// TODO(ryt): For Istanbul and versions after Istanbul, return CVM_VERSION_TWO
+	version := kernel.CVM_VERSION_ONE
+	if config.IsIstanbul(num) {
+		version = kernel.CVM_VERSION_TWO
+	}
+	return version
+}
+
+func (s *Synapse) InferByInfoHash(
+	modelInfoHash, inputInfoHash string,
+	cvmVersion int, cvmNetworkId int64) ([]byte, error) {
 	if s.config.IsRemoteInfer {
-		return s.remoteInferByInfoHash(modelInfoHash, inputInfoHash)
+		return s.remoteInferByInfoHash(modelInfoHash, inputInfoHash, cvmVersion, cvmNetworkId)
 	}
-	return s.inferByInfoHash(modelInfoHash, inputInfoHash)
+	return s.inferByInfoHash(modelInfoHash, inputInfoHash, cvmVersion, cvmNetworkId)
 }
 
-func (s *Synapse) InferByInputContent(modelInfoHash string, inputContent []byte) ([]byte, error) {
+func (s *Synapse) InferByInputContent(
+	modelInfoHash string, inputContent []byte,
+	cvmVersion int, cvmNetworkId int64) ([]byte, error) {
 	if s.config.IsRemoteInfer {
-		return s.remoteInferByInputContent(modelInfoHash, inputContent)
+		return s.remoteInferByInputContent(modelInfoHash, inputContent, cvmVersion, cvmNetworkId)
 	}
-	return s.inferByInputContent(modelInfoHash, inputContent)
+	return s.inferByInputContent(modelInfoHash, inputContent, cvmVersion, cvmNetworkId)
 }
 
-func (s *Synapse) GetGasByInfoHash(modelInfoHash string) (gas uint64, err error) {
+func (s *Synapse) GetGasByInfoHash(modelInfoHash string, cvmNetworkId int64) (gas uint64, err error) {
```

Comment:
Gas calculation is same between v1 and v2 , right?

Reply:
Yes